tests/test_tasks.py

```diff
@@ -800,8 +800,13 @@ async def foo():
 
         @flow
         async def my_flow():
-            future = await foo.submit()
-            state = await future.wait()
+            if PREFECT_EXPERIMENTAL_ENABLE_NEW_ENGINE:
+                future = foo.submit()
+                future.wait()
+                state = future.state
+            else:
+                future = await foo.submit()
+                state = await future.wait()
```

Comment:
Some of the tests differ between the current engine and the new engine due to the sync-only nature of the new task runner and `PrefectFuture` interfaces